File: projects/grind-75/python/binary/add_binary.py
class Solution(object):
    @classmethod
    def addBinary(self, a: str, b: str) -> str:
        res = ""
        carry = 0
        i = len(a) - 1
        j = len(b) - 1

        while i >= 0 or j >= 0 or carry > 0:
            n1 = int(a[i]) if i >= 0 else 0
            n2 = int(b[j]) if j >= 0 else 0

            total = n1 + n2 + carry
            carry = total // 2
            res = f"{total % 2}" + res

            i -= 1
            j -= 1

        return res

File: projects/grind-75/typescript/binary/add-binary.ts
export default function addBinary(a: string, b: string): string {
  let res = '';
  let carry = 0;
  let i = a.length - 1;
  let j = b.length - 1;

  while (i >= 0 || j >= 0 || carry > 0) {
    let n1 = i >= 0 ? Number(a[i]) : 0;
    let n2 = j >= 0 ? Number(b[j]) : 0;

    let total = n1 + n2 + carry;
    carry = Math.floor(total / 2);
    res = (total % 2) + res;

    // Move to the next digits
    i--;
    j--;
  }

  return res;
}
